lib/TestApp/Annotations/Controllers.cs

```diff
@@ -0,0 +1,13 @@
+﻿using static PrincipleStudios.OpenApiCodegen.Server.Mvc.TestApp.OneOf.PetControllerBase;
+
+namespace PrincipleStudios.OpenApiCodegen.Server.Mvc.TestApp.Annotations
+{
+	public class DogController : DogControllerBase
+	{
+		protected override Task<AddDogActionResult> AddDog(Dog addDogBody)
+		{
+			this.DelegateRequest(addDogBody);
+			return this.DelegateResponse<AddDogActionResult>();
+		}
+	}
```

Comment:
Looks like a good implementation, but I don't think I see any tests that calls this to ensure the annotations work the way we expect. Can you write a few positive/negative tests for those annotations?

Reply:
Tests added.